chrun.go

```diff
@@ -52,27 +52,33 @@ func pullImage(image string) {
 
 func chroot(root string, call string) {
 	//Hold onto old root
-	oldroot, err := os.Open("/")
+	oldrootHandle, err := os.Open("/")
 	if err != nil {
 		panic(err)
 	}
-	defer oldroot.Close()
+	defer oldrootHandle.Close()
 
+	//Create command
 	fmt.Printf("Running %s in %s\n", call, root)
 	cmd := exec.Command(call)
-	must(syscall.Chroot(root))
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+
+	//New Root time
+	must(syscall.Chdir(root))
+	must(syscall.Chroot(root))
+
 	err = cmd.Run()
 	if err != nil {
 		fmt.Println(err)
 	}
 
 	//Go back to old root
 	//So that we can clean up the temp dir
-	must(oldroot.Chdir())
+	must(syscall.Fchdir(int(oldrootHandle.Fd())))
```

Comment:
Why not must(syscall.Chdir(oldrootHandle.Name())) ? Name method returns the full path.